scripts/NuGetUtilities.psm1

```diff
@@ -0,0 +1,155 @@
+
+    if ($suffix -ne "") {
+        $arguments += "-Suffix", $Suffix
+    }
+
+    Write-CommandLine $NuGetExePath $arguments
+
+    & $NuGetExePath $arguments
+    if ($LASTEXITCODE -ne 0) {
+        Exit-WithFailureMessage $ScriptName "$project NuGet package creation failed."
+    }
+
+    Write-Information "  Successfully created package '$BinRoot\NuGet\$Configuration\$Project.$version.nupkg'."
+}
+
+function New-NuGetPackages($configuration, $projects) {
+    $version = Get-PackageVersion
+
+    # We can build the NuGet packages for library projects directly from their
+    # project file.
+    foreach ($project in $projects.NewLibrary) {
+        New-NuGetPackageFromProjectFile $configuration $project $version
+    }
+
+    # Unfortunately, application projects like MultiTool need to include things
+    # that are not specified in the project file, so their packages still require
+    # a .nuspec file.
+    foreach ($project in $Projects.NewApplication) {
+        New-NuGetPackageFromNuSpecFile $configuration $project $version
+    }
+}
+
+function Get-NuGetApiKey {
+    # We temporarily implement this function by parsing the key from the file
+    # SetNuGetJSarifApiKey.cmd, which is expected to exist in the parent
+    # directory of the developer's sarif-sdk enlistment.
+    # In future, we should get this key from the Azure Key Vault.
+    $apiKeyPath = Join-Path $PSScriptRoot ..\..\SetNuGetSarifApiKey.cmd
+    if (-not (Test-Path $apiKeyPath)) {
+        Exit-WithFailureMessage NuGetUtilties "API key file $apiKeyPath does not exist."
+    }
+
+    # Everything that isn't a double quote, followed by a double quote, followed
+    # by the key (everything up to the next double quote).
+    $pattern = '^[^"]*"(?<key>[^"]*)'
+
+    $firstLine = [IO.File]::ReadAllLines($apiKeyPath)[0]
+    if ($firstLine -match $pattern) {
+        $matches["key"]
+    } else {
+        Exit-WithFailureMessage NuGetUtilties "API key file $apiKeyPath does not contain a key."
+    }
+}
+
+function Set-NuGetApiKey {
+    $apiKey = Get-NuGetApiKey
+
+    $arguments = "SetApiKey", $apiKey, "-Source", $PackageSource, "-Verbosity", "quiet"
+    Write-CommandLine $NuGetExePath $arguments
+
+    & $NugetExePath $arguments
+    if ($LASTEXITCODE -ne 0) {
+        Exit-WithFailureMessage $ScriptName "Could not set NuGet API key."
+    }
+}
+
+function Hide-NuGetPackage($project, $version) {
+    $arguments = "delete", $project, $version, "-Source", $PackageSource
+    Write-CommandLine $NuGetExePath $arguments
+
+    & $NugetExePath $arguments
+    if ($LASTEXITCODE -ne 0) {
+        Exit-WithFailureMessage $ScriptName "Could not delist NuGet package $project $version."
+    }
+}
+
+function Hide-NuGetPackages {
+    Set-NuGetApiKey
+
+    $version = Get-PackageVersion -Previous
+    foreach ($project in $Projects.NewProduct) {
+        Hide-NuGetPackage $project $version
+    }
+}
+
+Export-ModuleMember -Function `
+    Hide-NuGetPackages, `
+    New-NuGetPackages
```

Comment:
>NuGetPackages [](start = 8, length = 13)

I exported fewer functions than Everett had previously exported -- just the ones that `BuildAndTest.cmd` and `DelistCurrentPackages.cmd` need. We can export more later if we need them.